platform/darwin/src/MGLMapSnapshotter.mm

```diff
@@ -117,180 +118,190 @@ - (void)startWithQueue:(dispatch_queue_t)queue completionHandler:(MGLMapSnapshot
     
     _loading = true;
     
-    dispatch_async(queue, ^{
-        _snapshotCallback = std::make_unique<mbgl::Actor<mbgl::MapSnapshotter::Callback>>(*mbgl::Scheduler::GetCurrent(), [=](std::exception_ptr mbglError, mbgl::PremultipliedImage image, mbgl::MapSnapshotter::Attributions attributions, mbgl::MapSnapshotter::PointForFn pointForFn) {
-            _loading = false;
+    __weak __typeof__(self) weakSelf = self;
+    _snapshotCallback = std::make_unique<mbgl::Actor<mbgl::MapSnapshotter::Callback>>(*mbgl::Scheduler::GetCurrent(), [=](std::exception_ptr mbglError, mbgl::PremultipliedImage image, mbgl::MapSnapshotter::Attributions attributions, mbgl::MapSnapshotter::PointForFn pointForFn) {
+        weakSelf.loading = false;
```

Comment:
This lambda uses `weakSelf` multiple times, with interspersed logic. This introduces a race condition: `self` could be freed at any time during the lambda’s execution, leading to undefined behavior. `weakSelf` should generally be accompanied by a `strongSelf` at the top of the lambda or block that strongly (but temporarily) captures the object. The only exception is for trivial blocks where `weakSelf` is used only once and the result of the statement is discarded.